mcgill-augmented-reality-frontend/screens/SignupScreen.js

```diff
@@ -0,0 +1,135 @@
+import { React, useState } from 'react';
+import { StyleSheet, View, TouchableOpacity, TouchableWithoutFeedback, Text, TextInput, Keyboard } from 'react-native';
+import { registerUser } from '../api';
+
+function SignupScreen({ navigation }) {
+
+    const [name, onChangeName] = useState('');
+    const [email, onChangeEmail] = useState('');
+    const [password, onChangePassword] = useState('');
+    const [isErrorShow, setIsErrowShow] = useState(false);
+
+    function handlePress() {
+        let valid = email.includes('@') && email.includes('.');
+
+        if (valid) {
+            registerUser(email, password)
+            .then(res => {
+                if (!res.ok) {
+                    console.log(res.error())
+                }
+            });
+            // navigation.navigate('nextScreenName');
+        } else {
+            setIsErrowShow(true);
+        }
+        
+    }
+
+    return (
+        <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+            <View style={styles.container}>
+                <Text style={styles.signupText}>Signup</Text>
+                <View style={styles.spaceSmall}/>
+                <Text style={styles.infoText}>
+                    Please fill out the following details to create an account
+                </Text>
+                <View style={styles.spaceLarge}/>
+                <TextInput 
+                    style={styles.input} 
+                    onChangeText={onChangeName}
+                    value={name}
+                    placeholder='Full Name' 
+                />
+                <View style={styles.spaceSmall}/>
+                <TextInput 
+                    style={styles.input} 
+                    onChangeText={onChangeEmail} 
+                    value={email}
+                    placeholder='Email' 
+                />
+                <View style={styles.spaceSmall}/>
+                <TextInput 
+                    style={styles.input} 
+                    onChangeText={onChangePassword}
+                    value={password} 
+                    placeholder='Password'
+                    secureTextEntry={true}
+                />
+                <View style={styles.spaceSmall}/>
+                <Text style={{
+                    color: 'red',
+                    fontSize: 15,
+                    textAlign: 'center',
+                    width: '80%',
+                    opacity: isErrorShow ? 1 : 0
+                }}>
+                    Please enter a valid email.
+                </Text>
+                <View style={styles.spaceLarge}/>
```

Comment:
We'll need the other info the db now for the sign up. student id, faculty, year, ...